File: backend/ollama_client.py
import httpx
import json

OLLAMA_URL = "http://localhost:11434/api"

class OllamaClient:
    @staticmethod
    async def get_response_stream(model: str, prompt: str, context: list = []):
        payload = {
            "model": model,
            "prompt": prompt,
            "context": context,
            "stream": True
        }
        async with httpx.AsyncClient(timeout=None) as client:
            async with client.stream("POST", f"{OLLAMA_URL}/generate", json=payload) as response:
                async for line in response.aiter_lines():
                    if line:
                        data = json.loads(line)
                        yield f"data: {json.dumps(data)}\n\n"
                        if data.get("done"):
                            break

    @staticmethod
    async def list_models():
        async with httpx.AsyncClient() as client:
            try:
                resp = await client.get(f"{OLLAMA_URL}/tags", timeout=5.0)
                resp.raise_for_status()
                return resp.json().get("models", [])
            except httpx.RequestError:
                return [] # Return empty list if Ollama is offline

File: backend/routes/ollamaClient.js
const axios = require('axios');

const OLLAMA_API_URL = 'http://localhost:11434/api'; // Update with your Ollama server URL

async function listModels() {
  try {
    const response = await axios.get(`${OLLAMA_API_URL}/tags`);
    return response.data.models;
  } catch (error) {
    console.error('Error listing models:', error);
    throw error;
  }
}

async function generateResponse(model, prompt, context = []) {
  try {
    const payload = { model, prompt, context };
    const response = await axios.post(`${OLLAMA_API_URL}/generate`, 
payload);
    return response.data;
  } catch (error) {
    console.error('Error generating response:', error);
    throw error;
  }
}

// Add more functions for session management and model switching as needed

module.exports = { listModels, generateResponse };
